### PR TITLE
EPMRPP-34019: adds demoDataGenerator with redux-form

### PR DESCRIPTION
**Pull request by pawellesniowski**

### CODE REVIEW

File: app/src/pages/inside/settingsPage/demoDataTab/demoDataGenerator.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  .input-wrapper {
+    width: 310px;
+    height: 30px;
+    margin-bottom: 20px;
+    @media (max-width: $SCREEN_XS_MAX) {
+      display: none;
+    }
+  }
+
+  .buttonGenerate {
```
Comment (IhorSyerkov) [inline app/src/pages/inside/settingsPage/demoDataTab/demoDataGenerator.scss]: use kebab-case for naming

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+import styles from './demoDataGenerator.scss';
+import warningSign from './img/ic-warning.svg';
+import DemoDataGeneratorForm from './demoDataGeneratorForm';
+
+const cx = classNames.bind(styles);
+
+export class DemoDataGenerator extends Component {
+  generateDemoData = (values) => {
+    console.log('values: ', values);
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx]: Remove console log & add fetch request

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        <p className={cx('mobile-hint')}>
+          <FormattedMessage
+            id={'DemoDataGenerator.mobileHint'}
+            defaultMessage={'You can generate data only on desktop view.'}
+          />
+        </p>
+        <h5 className={cx('description-header')}>
+          <FormattedMessage
+            id={'DemoDataGenerator.descriptionHeader'}
+            defaultMessage={'THE SYSTEM WILL GENERATE THE FOLLOWING DEMO DATA:'}
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx]: Use css to transform text to uppercase.

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss [hunk 4]
```diff
@@ -0,0 +1,93 @@
+.demo-data-generator-container {
+  padding: 20px 10px;
+  display: flex;
+  flex-direction: column;
+
+  .description-header {
+    margin-top: 15px;
+    font-size: 12px;
+    font-weight: $FONT-SEMIBOLD;
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss]: font-family

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+  .warning-block {
+    .warning-sign {
+      display: inline-block;
+      margin-right: 5px;
+    }
+    .warning-text {
+      display: inline-block;
+      font-size: 13px;
+      font-weight: $FONT-SEMIBOLD;
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss]: font-family

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+import { BigButton } from 'components/buttons/bigButton';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+
+import styles from './demoDataGenerator.scss';
+
+const cx = classNames.bind(styles);
+
+const DemoDataGeneratorForm = (props) => {
+  const { handleSubmit } = props;
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx]: const DemoDataGeneratorForm = ({ handleSubmit }) =>

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+      </div>
+    </form>
+  );
+};
+
+DemoDataGeneratorForm.propTypes = {
+  ...propTypes,
+};
+
+export default reduxForm({
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx]: add as decorator

**Comment by ArtsiomYeliseyenka** [comment]
Additionally there is no functionality related input disabling and preloader appearing in case when demo data is generating

File: app/src/pages/inside/settingsPage/demoDataGenerator/Button.jsx [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import React from 'react';
+
+export const Button = () => <button />;
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/Button.jsx]: What is this for?

File: app/src/pages/inside/settingsPage/demoDataGenerator/README.md [hunk 9]
```diff
@@ -0,0 +1 @@
+## **demoDataGenerator**
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/README.md]: What is this for?

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+      font-size: 13px;
+      color: $COLOR--charcoal-grey;
+      line-height: 1.85;
+      display: flex;
+      flex-direction: column;
+      justify-content: center;
+    }
+    @media (max-width: $SCREEN_XS_MAX) {
+
+    }
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.scss]: Remove if no needed

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.stories.jsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+ * along with Report Portal.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { action } from '@storybook/addon-actions';
+import { Button } from './Button';
+// import { DemoDataGenerator } from './demoDataGenerator';
+
+storiesOf('Button', module)
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.stories.jsx]: Do not include your test & learn files into pull requests. Moreover, we write stories only for common reusable components.

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+
+import styles from './demoDataGenerator.scss';
+
+const cx = classNames.bind(styles);
+
+@reduxForm({
+  form: 'postfix',
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx]: Incorrect form name. 'postfix' it's a field name. Form should named something like 'demoDataGeneratorForm'

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx [hunk 13]
```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+import { propTypes, reduxForm } from 'redux-form';
+import { Input } from 'components/inputs/input';
+import classNames from 'classnames/bind';
+import { FormattedMessage } from 'react-intl';
+
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx]: remove blank line

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+          <BigButton type="submit" disabled={loading}>
+            <span className={cx('generate-demo-data')}>
+              <FormattedMessage
+                id={'DemoDataGeneratorForm.buttonGenerate'}
+                defaultMessage={'Generate Demo Data'}
+              />
+            </span>
+          </BigButton>
+        </div>
+        {loading
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGeneratorForm.jsx]: loading &&

File: app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx [hunk 15]
```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import classNames from 'classnames/bind';
+
```
Comment (ArtsiomYeliseyenka) [inline app/src/pages/inside/settingsPage/demoDataGenerator/demoDataGenerator.jsx]: remove blank lines between imports

File: app/src/controllers/projectSettings/demoData/actionCreators.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
+import { GENERATE_DEMO_DATA } from './constans';
```
Comment (ArtsiomYeliseyenka) [inline app/src/controllers/projectSettings/demoData/actionCreators.js]: misprint in constan**t**s.js file name

File: app/src/controllers/projectSettings/demoData/sagas.js [hunk 17]
```diff
@@ -0,0 +1,2 @@
+
+
```
Comment (ArtsiomYeliseyenka) [inline app/src/controllers/projectSettings/demoData/sagas.js]: No functionality implemented.

**Pull request closed without merging**